src/components/Application/ApplicationSummary.tsx

```diff
@@ -1,57 +1,49 @@
 import React from 'react'
-import { Button, Container, Form, Grid, Header, Input, Label, Segment } from 'semantic-ui-react'
+import { Button, Container, Form, Grid, Header, Input, Segment } from 'semantic-ui-react'
+import { ApplicationResponse, TemplateElementCategory } from '../../utils/generated/graphql'
 import {
-  ApplicationResponse,
-  TemplateElement,
-  TemplateElementCategory,
-} from '../../utils/generated/graphql'
-import useLoadSummary from '../../utils/hooks/useLoadSummary'
-import { ElementAndResponse, TemplateSectionPayload } from '../../utils/types'
-import Loading from '../Loading'
+  ApplicationElementState,
+  ElementState,
+  ResponseFull,
+  ResponsesFullByCode,
+  TemplateSectionPayload,
+} from '../../utils/types'
 
 interface ApplicationSummaryProps {
-  applicationId: number
-  sections: TemplateSectionPayload[]
+  templateSections: TemplateSectionPayload[]
+  elementsState: ApplicationElementState
+  responsesByCode: ResponsesFullByCode
   onSubmitHandler: () => void
 }
 
+interface SectionGroups {
+  [key: string]: ApplicationElementState
+}
+
 const ApplicationSummary: React.FC<ApplicationSummaryProps> = ({
-  applicationId,
-  sections,
+  templateSections,
+  elementsState,
+  responsesByCode,
   onSubmitHandler,
 }) => {
-  const { sectionElements, error, loading } = useLoadSummary({
-    applicationId,
-    sectionIds: sections.map(({ id }) => id),
-  })
-
-  const getQuestion = (question: TemplateElement, response: ApplicationResponse | null) => {
-    return question.category === TemplateElementCategory.Question ? (
-      <Form.Field>
-        <Header as="h3" content={question.title} />
-        <Input disabled error={!response}>
-          {response?.value}
-        </Input>
-      </Form.Field>
-    ) : question.elementTypePluginCode !== 'pageBreak' ? (
-      <Form.Field>
-        <Header as="h5" content={question.title} />
-      </Form.Field>
-    ) : null
-  }
+  const sectionsAndElements = Object.entries(elementsState).reduce(
+    (sectionGroups: SectionGroups, [code, item]) => {
+      const group = sectionGroups[item.section] || {}
+      group[code] = item
+      sectionGroups[item.section] = group
+      return sectionGroups
+    },
```

Comment:
I think if we are doing preliminary reshaping like this, it would be great to put all of the required info in the new object (so we do the minimum amount of logic in jsx)

Also it maybe easier to deal with arrays (rather than objects) in jsx.

i.e. if sectionsAndElements was:

```
[
 {
  section: {full section info}
  elements: [ 
      {
          element: {code etc}
          value: {  }
       }
   ]
 }
]
```

It's also easier to sort arrays, by the way are the elements sorted by index within the section ?

Reply:
Doing this now!

Reply:
Have added this suggestion. Please have a look see if you like. I have moved to use arrays as you suggested. And I'm using a sort on the sections by index to keep in the correct order.

Reply:
looks great thanks !